### PR TITLE
test: fixing one tsan issue with base_client_integration_test

### PR DESCRIPTION
**Pull request by alyssawilk**
Risk Level: low
Testing: yes
Docs Changes: n/a
Release Notes: n/a

### CODE REVIEW

File: mobile/test/common/integration/base_client_integration_test.h [hunk 1]
```diff
@@ -74,6 +74,7 @@ class BaseClientIntegrationTest : public BaseIntegrationTest {
   Platform::StreamSharedPtr stream_;
   Platform::EngineSharedPtr engine_;
```
Comment (RyanTheOptimist) [inline mobile/test/common/integration/base_client_integration_test.h:75]: Can we add a GUARDED_BY annotation to enforce that this must only be touched when the lock is held?

**Review by RyanTheOptimist — COMMENTED** [review]
Oh, great catch! Thanks for finding this.

File: mobile/test/common/integration/base_client_integration_test.h (hunk 1 above)
Comment (alyssawilk) [reply mobile/test/common/integration/base_client_integration_test.h:75]: yeah the only cross-thread issue was during start-up but I can lock all other access on principle.

**Review by RyanTheOptimist — APPROVED** [review]
Thanks for doing this!

**Review by RyanTheOptimist — APPROVED** [review]

**Review by RyanTheOptimist — APPROVED** [review]

**Pull request merged**